pkg/operator/ceph/object/objectstore.go

```diff
@@ -954,40 +957,109 @@ func zoneUpdateWorkaround(objContext *Context, expectedZone, gotZone map[string]
 	if err != nil {
 		return err
 	}
+
```

Comment:
This method is a bit complicated. According to the comment above, this fix is only needed for v18.2.2 and older. If we just require v18.2.3 and newer for using the placement pools feature, can we get rid of this method now? For new features like this, it's reasonable IMO to require a newer ceph release.

Reply:
This method (`zoneUpdateWorkaround`) was introduced for SharedPools feature so it also will be called when only shared pools are set:
```yaml
spec:
  sharedPools:
    dataPoolName: data-pool
    metadataPoolName: meta-pool
```
I can't remove it if someone is using shared pools with < v18.2.2. Should i try to refactor it?

Reply:
Oh right, I was the one who added it! :) 
So the workaround was only related to modifying the shared pool settings. Your workaround for removing the default placement is likely not related to that version comment, so we likely need to move that comment from the function header down to line 1034. And we will need to keep your workaround longer than reef, right?

Reply:
yes, we should keep it longer. i can move this workaround to a separate function but in this case i will have to duplicate a lot of code because both workarounds are aditing pool placements in the same json.
will it be ok if i just change comment?

Reply:
Agreed, sounds good to just change the comments so we can avoid any code duplication

Reply:
done